x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/Pattern.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.ql.expression.predicate.regex;
+
+interface Pattern {
```

Comment:
Maybe `RegexPattern`

Reply:
I'm not sure, the LikePattern is not 100% regex pattern.

Reply:
How about `StringMatchPattern`?

Reply:
Why not `StringPattern` ?

Reply:
sure that too!